pcsx2/gui/AppMain.cpp

```diff
@@ -529,11 +531,34 @@ extern bool FMVstarted;
 extern bool renderswitch;
 extern bool EnableFMV;
 
-void DoFmvSwitch()
+void DoFmvSwitch(bool on)
 {
-	ScopedCoreThreadPause paused_core( new SysExecEvent_SaveSinglePlugin(PluginId_GS) );
-	renderswitch = !renderswitch;
-	paused_core.AllowResume();
+	if (g_Conf->GSWindow.IsToggleAspectRatioSwitch) {
+		AspectRatioType& art = g_Conf->GSWindow.AspectRatio;
+		if (!EnableFMV && !FMVstarted && art == AspectRatio_4_3 && prevAR != art)
+			prevAR = art;
+
+		if (on) {
+			if (art != AspectRatio_4_3) {
+				prevAR = art;
+				art = AspectRatio_4_3;
+			}
+		}
+		else if (!on) {
+			if (art == AspectRatio_4_3 && prevAR != AspectRatio_4_3)
+				art = prevAR;
+		}
+
```

Comment:
Anyway, that is a bad solution.
Keep
if (on) {
 do thing
} else {
 do other thing
}
It isn't a good idea to merge other thing into an else if because it is 2 different/independent actions (It is hard to explain).

elseif is better for
if (a == 0)
elseif (a == 1)
else